authentication with Firebase Auth and JWT

EvelynBunnyDev opened the pull request:
# Authentication with Firebase Auth and JWT

## :recycle: Current situation & Problem
Please refer to these two discussion posts for context: 
https://github.com/orgs/CS342/discussions/78.
https://github.com/orgs/CS342/discussions/83


## :gear: Release Notes 
Modifications has been made to Prisma/Chat/ChatView
1. We generated JWT token in app in the `signInWithFirebase` function: `token = try await Auth.auth().currentUser?.getIDToken()`.
2. Then, the token is sent to the frontend via a URL parameter: `if let url = URL(string: "http://localhost:3000?token=\(token)")`
Successful implementation is reflected by passing a JWT token from Swift to the web chat interface, which connects to the python backend and prints "user successfully authenticated via JWT". 

## :books: Documentation
See in-line documentation

## :white_check_mark: Testing
This codebase has not been thoroughly tested. We will address the stylistic requirements later.


## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/CS342/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/CS342/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/CS342/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/CS342/.github/blob/main/CONTRIBUTING.md).

bryant-jimenez committed:
Client handlers for notification received timestamp

Handlers for receiving a notification in the background or foreground, call methods in the PrismaStandard for setting the time received in Firestore.

EvelynBunnyDev committed:
update ChatView to incorporate JWT token auth

EvelynBunnyDev committed:
Update ChatView.swift

EvelynBunnyDev committed:
updates to firebase sign-in

EvelynBunnyDev committed:
Update ChatView.swift

mjoerke commented on Prisma/Chat/ChatView.swift:
```diff
@@ -36,6 +63,53 @@ struct ChatView: View {
+        }
+    }
+    
+    func sendTokenToBackend(token: String, completion: @escaping (String?) -> Void) {
```
Why is this function needed? I thought the frontend would take the token from the URL parameter and use that to authenticate the frontend <-> backend connection.

bryant-jimenez committed:
updat package dependencies

EvelynBunnyDev committed:
Update ChatView.swift

bryant-jimenez committed:
Conforming to Spezi 1.2 Notification handling

PSchmiedmayer committed:
Fix Push Notifications

EvelynBunnyDev committed:
Merge remote-tracking branch 'origin/bryant/notification-status-to-firebase' into jwt-auth

EvelynBunnyDev committed:
passing JWT token from iOS to web

updating the packages, ChatView adding getFirebaseIDToken, updating the GOogleService plist.

EvelynBunnyDev committed:
updating feature flags to not use the emulator

EvelynBunnyDev committed:
enable firebase emulator

dhruvna1k committed:
Merge branch 'main' into jwt-auth

dhruvna1k committed:
swiftlint fixes

dhruvna1k committed:
Merge branch 'main' into jwt-auth

dhruvna1k committed:
swiftline unwrapping

dhruvna1k committed:
fixed google plist adding
